lib/control.ml

```diff
@@ -30,11 +30,12 @@ let check_for_interrupt () =
 
 (** This function does not work on windows, sigh... *)
 let unix_timeout n f x =
+  let open Unix in
   let timeout_handler _ = raise Timeout in
   let psh = Sys.signal Sys.sigalrm (Sys.Signal_handle timeout_handler) in
-  let _ = Unix.alarm n in
+  let _ = setitimer ITIMER_REAL {it_interval = 0.; it_value = n} in
   let restore_timeout () =
-    let _ = Unix.alarm 0 in
+    let _ = setitimer ITIMER_REAL { it_interval = 0.; it_value = 0. } in
```

Comment:
OCaml documentation for `setitimer` says "On windows: not implemented"

So this may need a bit more work.

Reply:
Indeed, but this was also true for `alarm`. For windows, there is already the separately implemented `windows_timeout`, which I also trivially converted to support high resulution.

Reply:
Oh indeed :facepalm: